static/app.js

```diff
@@ -16,20 +16,28 @@ app.controller('appCtrl', function ($http, $mdSidenav, $mdDialog, $location) {
       templateUrl: 'static/partials/transitmap.html',
       parent: angular.element(document.body),
       clickOutsideToClose:true,
+      controller: DialogController,
       fullscreen: true
     });
   };
 
 });
 
+function DialogController($scope, $mdDialog) {
+  $scope.cancel = function() {
+    $mdDialog.cancel();
+  };
+}
+
 
 // configure our routes
 app.config(function($routeProvider) {
 	$routeProvider
 
 		.when('/trains/:route_name', {
 			templateUrl : 'static/partials/route.html',
-			controller  : 'routeController'
+			controller: '',
```

Comment:
Because I had the controller defined in the html it was instantiating it twice >.<